cmd/topicctl/subcmd/reset.go

```diff
@@ -100,10 +115,24 @@ func resetOffsetsRun(cmd *cobra.Command, args []string) error {
 	}
 	partitionOffsets := map[int]int64{}
 
-	if len(resetOffsetsConfig.partitions) > 0 {
+	if len(resetOffsetsConfig.partitionOffsetMap) > 0 {
+		for partition, offset := range resetOffsetsConfig.partitionOffsetMap {
+			var partitionID int
+			if partitionID, err = strconv.Atoi(partition); err != nil {
+				return fmt.Errorf("partition value %s must be a number", partition)
+			}
+			if _, ok := partitionIDsMap[partitionID]; !ok {
+				return fmt.Errorf("partition %d not found in topic %s", partitionID, topic)
+			}
+
+			partitionOffsets[partitionID] = offset
+
+		}
+
+	} else if len(resetOffsetsConfig.partitions) > 0 {
 		for _, partition := range resetOffsetsConfig.partitions {
 			if _, ok := partitionIDsMap[partition]; !ok {
-				return fmt.Errorf("Partition %d not found in topic %s", partition, topic)
```

Comment:
Can we revert the lowercasing of these errors? We use uppercase widely in this repo

Reply:
addressed in [bd7b969](https://github.com/segmentio/topicctl/commit/bd7b96943fcfd3d08b1ad4d1052be442883c7642)